extendr-api/src/wrapper/promise.rs

```diff
@@ -82,9 +87,13 @@ impl Promise {
 
 impl std::fmt::Debug for Promise {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("Promise")
-            .field("code", &self.code())
-            .field("environment", &self.environment())
-            .finish()
+        let mut result = f.debug_struct("Promise");
+
+        #[cfg(feature = "non-api")]
+        {
+            let result = result.field("code", &self.code());
+            let result = result.field("environment", &self.environment());
```

Comment:
doesnt this need to be `mutate = ` and not `let result ` as the inner result is scoped to that section? Or does the outer ` { } ` get ignored because of the cfg flag?

Reply:
`cfg`applies to the `{}`, so we are good in that case :)